src/main/java/com/example/dog/Dog.java

```diff
@@ -1,38 +1,50 @@
 package com.example.dog;
 
 import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 
 import javax.validation.constraints.*;
-import java.util.Calendar;
+import java.time.ZonedDateTime;
 
 import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.ANY;
 import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.NONE;
 
 @JsonAutoDetect(fieldVisibility = ANY, getterVisibility = NONE, isGetterVisibility = NONE)
 public class Dog {
-    @NotNull
-    @Size(min = 1, max = 100, message = "Name must be between 1 and 100 characters long.")
+
+    private int id;
+
+    @NotNull(message = "Name cannot be null")
+    @Size(min = 1, max = 100, message = "Name must be between 1 and 100 characters long")
     private String name;
 
-    @NotNull
-    @Min(value = 0, message = "Height must be equal or greater than 0.")
+    @Min(value = 1, message = "Height must be greater than 0, was ${validatedValue}")
     private int height;
 
-    @NotNull
-    @Min(value = 0, message = "Weight must be equal or greater than 0.")
+    @Min(value = 1, message = "Weight must be greater than 0, was ${validatedValue}")
     private int weight;
 
-    @Past(message = "Birthday must be before NOW.")
-    private Calendar birthday;
+    @JsonDeserialize(using = ZonedDateTimeDeserializer.class)
```

Comment:
Jackson has its own standard implementation for the ZonedDateTime (de)serialization.

Reply:
Jackson deserializes it incorrectly. The `dateOfBirth.zone.id` is always "UTC".

Reply:
Well, I wouldn't go as far as calling this "incorrect".. It does pass a correct instant. But looses the original timezone information. If this is important, then it's ok to have your own implementation, but you should leave JavaDocs explaining why you wrote it.